com.blitterstudio.amiberry.yml

```diff
@@ -0,0 +1,98 @@
+app-id: com.blitterstudio.amiberry
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+command: amiberry.sh
+rename-icon: amiberry
+copy-icon: true
+finish-args:
+  - --device=all
+  # X11 + XShm access
+  - --share=ipc
+  # Wayland access
+  - --socket=wayland
+  - --socket=fallback-x11
+  # Needs to talk to the network:
+  - --share=network
+  # Allow bluetooth (e.g. controllers)
+  - --allow=bluetooth
+  # Play sounds using PulseAudio
+  - --socket=pulseaudio
+  # Filesystem access
+  - --filesystem=host
```

Comment:
Why do you need full host?

Reply:
The app has several features, including read/write to disk (for using hardfiles or disk images), but also a unique feature that allows the emulated environment to (optionally) send commands to the host, to be executed. This allows things like triggering a host application to handle a filetype that the emulated system couldn't handle (like an mkv file, for instance. The emulated Amiga cannot play it back, but it could call VLC from the host to do that). I don't think these would work without full host access, right?

Reply:
> emulated environment to (optionally) send commands to the host, to be executed

You won't be able to run arbitrary commands from inside the sandbox. What commands are these specifically?

> including read/write to disk 

Are there any specific locations?

Reply:
The commands could be anything, they are sent to a shell directly and are executed under the user context. I know the sandbox will limit what's available, of course.
Regarding locations, there are no hardcoded ones: there are file and directory requesters to allow  people to choose where to load an image from, or where to create a new hardfile if needed. We could have a convention to say that those _should_ be under the flatpak's `XDG_DATA_HOME` dir, perhaps?

Reply:
the it doesn't do what you think it does.

Reply:
> load an image from

This would usually require at least access to xdg-pictures or home. Does this go through a filechooser? If so, do you support the filchooser portal https://flatpak.github.io/xdg-desktop-portal/docs/#gdbus-org.freedesktop.portal.FileChooser? Then you don't need any filesystem access for that.

> where to create a new hardfile if needed

Same as above. But if it doesn't, you need filesystem access.

> should be under the flatpak's XDG_DATA_HOME dir

That's usually for persistent application data. Without knowing what kind of files they are, it's hard to say where they should be saved.

Reply:
Thanks for the explanation!
The files the app usually deals with, are:
- It's own data files, which include a directory for config files, controller mappings, kickstart ROMs (which the user needs to copy in there, since they cannot be included due to copyright), etc. All these are placed under `XDG_DATA_HOME` and from my tests, they work fine there.
-  _Floppy Disk_ images (not actual pictures), which are copies from floppy disks of the past, sometimes compressed as `.zip` or `.lha` archives. 
- Hardfiles, which are hard drive images (like `.hdf` or `.chd`, similar to what M.A.M.E. also supports)
- Directories treated as virtual hard drives, where the files and directories are available to the Linux host, but they are read from the emulator as a virtual Amiga hard drive.
- There is an option to create new Hardfile or empty Disk Images, so that the user can fill it in with their own data.

All these go through File/Directory requesters within the app (not system ones), so I don't think the portal would work (at least, not without extensive code changes). The location these can be in is not hard coded anywhere, so they can exist anywhere that is accessible really. Moreover, the user might already have a collection of such files they want to use with the emulator, which will most likely exist outside the sandbox.

I'm open to suggestions on how this should be handled best. :)

Reply:
> I'm open to suggestions on how this should be handled best. :)

Not really a way to handle it, other than keeping it as is. Some of it is application's own data that should go in `XDG_DATA_HOME`, some of it is user facing that could be anywhere in home.